UI - Navbar for MapKnitter

stefannibrasil opened the pull request:
Related to #327 

Hi all,
My proposal here is to change the navbar to use the same layout as Public Lab. That way, whenever we need to create a new navbar for a new project, we can follow this pattern. This also prevents having different CSS files and classes for every project, letting us focus on the project layout.

I still need to fix some small issues, like the login form and the search form.

But I have some questions that I wanted to ask before moving on:

* The app is not using Bootstrap at all? I couldn't find a link to it, so I added a new one for Bootstrap 4. I may be missing something here, but it does not look like we are using Bootstrap. I am totally for using 
Bootstrap 4, they have some nice cards and mobile utilities that make our work easier.

* I noticed that the Resources link on the navbar was being hidden. Is it supposed to be removed from it?

Let me know your thoughts, please! @jywarren 

Some screenshots: to help with the review:

![screen shot 2019-02-25 at 11 57 11](https://user-images.githubusercontent.com/10670581/53364839-908f5900-38f4-11e9-998f-ef75d699907e.png)
![screenshot_2019-02-25 mapknitter](https://user-images.githubusercontent.com/10670581/53364841-908f5900-38f4-11e9-8377-7d2dba45b3e3.png)

jywarren commented:
Hi!!! So, like in plots2, the stylesheets are included here: https://github.com/publiclab/mapknitter/blob/master/app/views/layouts/application.html.erb#L16

bootstrap is included, and we seem to be on Bootstrap 3.2 -- see:

https://github.com/publiclab/mapknitter/blob/master/app/assets/stylesheets/application.css#L12

https://github.com/publiclab/mapknitter/blob/master/bower.json#L6

We should move to Yarn, see https://github.com/publiclab/mapknitter/pull/357 for progress on this, which should simplify things a bit. 

+1 this change, the menubar would be nice if standardized here!

stefannibrasil commented:
ahhh, thanks!!! So we are using Bootstrap 3.2.0 
https://github.com/publiclab/mapknitter/blob/master/bower.json#L6

I'll update the changes then. I'm not familiar with yarn and bower, but if that will simplify, awesome! :tada

I will try to finish today (tomorrow for sure), then I'll upload more detailed screenshots for this. Thanks for the help, @jywarren

jywarren commented:
But we should look at upgrading to bootstrap 4!! Shouldn't be too hard
compared to plots2 since this is such a simpler app!

On Thu, Feb 28, 2019, 3:26 PM Stefanni <notifications@github.com> wrote:

> ahhh, thanks!!! So we are using Bootstrap 3.2.0
> https://github.com/publiclab/mapknitter/blob/master/bower.json#L6
>
> I'll update the changes then. I'm not familiar with yarn and bower, but if
> that will simplify, awesome! :tada
>
> I will try to finish today (tomorrow for sure), then I'll upload more
> detailed screenshots for this. Thanks for the help, @jywarren
> <https://github.com/jywarren>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/360#issuecomment-468424980>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ--Y_kEjZSKbvdM5LpotWTYSwoxKks5vSDtUgaJpZM4bQpbw>
> .
>

stefannibrasil commented:
I'm totally for it! Maybe creating an issue for now? After this navbar, I can take a look.

jywarren commented:
sounds great!

On Thu, Feb 28, 2019 at 4:01 PM Stefanni <notifications@github.com> wrote:

> I'm totally for it! Maybe creating an issue for now? After this navbar, I
> can take a look.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/360#issuecomment-468436552>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ1J1eDp86Nds0JVkM9MPk__c0vevks5vSEONgaJpZM4bQpbw>
> .
>

stefannibrasil commented:
okay, I created this new issue #377  for the Bootstrap update. If you want to add it to projects page and add labels to it, it would be nice, I couldn't add any :+1:

stefannibrasil commented:
Hi @jywarren I changed the navbar to use Bootstrap 3.2.0 for now.

Here are some screenshots. One thing that I noticed was that the navbar was not using Bootstrap's collapse, so I changed that, no need for adding CSS for this behavior.

stefannibrasil commented:
![screenshot_2019-03-02 mapknitter 2](https://user-images.githubusercontent.com/10670581/53675375-8fd72980-3c4a-11e9-90c3-96fbb167d35d.png)


When I take screenshots the navbar gets in the middle of the map, IDK why!
Here I moved the search form to the navbar, but I left the previous one there for now.

stefannibrasil commented:
![screenshot_2019-03-02 mapknitter](https://user-images.githubusercontent.com/10670581/53675406-c14ff500-3c4a-11e9-90b5-3b2122c96a41.png)

this one looks better to see the navbar alone.

stefannibrasil commented:
Mobile:


![screen shot 2019-03-01 at 17 39 13](https://user-images.githubusercontent.com/10670581/53675415-d7f64c00-3c4a-11e9-8951-3e4aded3e387.png)



Opening the user dropdown:


![screen shot 2019-03-01 at 17 39 23](https://user-images.githubusercontent.com/10670581/53675417-ddec2d00-3c4a-11e9-978e-b15a914c537d.png)

stefannibrasil commented:
Maps page:

![screen shot 2019-03-01 at 17 38 53](https://user-images.githubusercontent.com/10670581/53675432-fd835580-3c4a-11e9-96c2-287dc764dde9.png)
![screenshot_2019-03-02 mapknitter](https://user-images.githubusercontent.com/10670581/53675433-fd835580-3c4a-11e9-8a7a-d96a19698435.png)

stefannibrasil commented:
Let me know what do you think! It leaves more space for adding contents later, I think :)

jywarren commented:
Ooh it's really nice! Is it ready for a merge? Thanks!!!

jywarren commented:
It looks like Travis is OK now?

stefannibrasil commented:
yeah, it's ready for a merge! Glad you liked :)
I think after I switched to normal PR travis ran as expected, so it's okay now.

After we merge this, should I start on updating Bootstrap version? Thanks :v:

jywarren commented on app/assets/stylesheets/header.css.scss:
```diff
@@ -1,130 +1 @@
-    font-family: 'Junction Light';
-  }
-
-  img.cat {
```
Did you find the easter egg???????

jywarren commented:
🎉 !!!! This is awesome. Thanks @stefannibrasil -- yeah that'd be great!
